### ISSUE TITLE
Protocols: enable individual fish protocol counts

### ISSUE THREAD

**Issue by ericbrasseur-NOAA** [nwfsc-fram/boatnet#409]
Many WCGOP protocols change requirements after a specific number of fish have been sampled.
Ex Eulachon protocol is 1 whole fish saved and 3 fish lengthed.  The whole fish should be randomly selected from the 3.  Enabling random  selection for this protocol would good as well.  The whole specimen trumps quantity so if there is less then 3 fish a whole fish will still always be saved.  Reference the count from the species comp and if greater than 1 enable a random assignment of the whole fish where random selection made from (protocol requirement -fish count) if less than protocol.

Some protocols are simple length and sex the first 5 fish then length only the next 20.

Basically realtime error checking should take into account if there are enough fish to do each step of the protocol, and stop warning for parts of the protocol that have been satisfied.
ie for 5s/l and 25 l (include the first 5)
warn incomplete if missing sex or length for the first 5 and only length for 6-25.
Protocols all contain the required information.
See optecs warnings for current methods.
